Tests/UnitTests/Shared/Shared.cpp

```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
 }
 
-int RunTests(Babylon::Polyfills::Console::CallbackT consoleCallback)
+TEST(JS, JSTests)
 {
     std::promise<int32_t> exitCode;
 
     Babylon::AppRuntime runtime{};
 
-    runtime.Dispatch([&exitCode, consoleCallback = std::move(consoleCallback)](Napi::Env env) mutable
-    {
+    runtime.Dispatch([&exitCode, consoleCallback = std::move([](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+        fprintf(stdout, "[%s] %s", EnumToString(logLevel), message);
+        fflush(stdout);
+    })](Napi::Env env) mutable {
         Babylon::Polyfills::XMLHttpRequest::Initialize(env);
         Babylon::Polyfills::Console::Initialize(env, std::move(consoleCallback));
         Babylon::Polyfills::Scheduling::Initialize(env);
         Babylon::Polyfills::URL::Initialize(env);
         Babylon::Polyfills::AbortController::Initialize(env);
         Babylon::Polyfills::WebSocket::Initialize(env);
 
-        env.Global().Set("SetExitCode", Napi::Function::New(env, [&exitCode](const Napi::CallbackInfo& info)
-        {
-            Napi::Env env = info.Env();
-            exitCode.set_value(info[0].As<Napi::Number>().Int32Value());
-        }, "SetExitCode"));
+        env.Global().Set("SetExitCode", Napi::Function::New(
+                                            env, [&exitCode](const Napi::CallbackInfo& info) {
+                                                Napi::Env env = info.Env();
+                                                exitCode.set_value(info[0].As<Napi::Number>().Int32Value());
+                                            },
+                                            "SetExitCode"));
 
         env.Global().Set("hostPlatform", Napi::Value::From(env, JSRUNTIMEHOST_PLATFORM));
     });
 
     Babylon::ScriptLoader loader{runtime};
-    loader.Eval("var global = {};", ""); // Required for chai.js
+    loader.Eval("var global = {};", "");             // Required for chai.js
     loader.Eval("var location = { href: '' };", ""); // Required for mocha.js
     loader.LoadScript("app:///Scripts/chai.js");
     loader.LoadScript("app:///Scripts/mocha.js");
     loader.LoadScript("app:///Scripts/tests.js");
 
-    return exitCode.get_future().get();
+    auto exit{exitCode.get_future().get()};
+
+    EXPECT_EQ(exit, 0);
+}
+
+TEST(Console, ConsoleLog)
+{
+    std::promise<int32_t> exitCode;
+
+    Babylon::AppRuntime runtime{};
+
+    runtime.Dispatch([&exitCode, consoleCallback = std::move([](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+        static size_t currIdx = 0;
+        static std::vector<const char*> messagesToTest = {"foo bar\n"};
+        const char* test = messagesToTest[currIdx++];
+        if (strcmp(message, test))
+        {
+            fprintf(stdout, "Expected: %s, received: %s", test, message);
+            ADD_FAILURE();
+        }
+
+        fprintf(stdout, "[%s] %s", EnumToString(logLevel), message);
+        fflush(stdout);
+    })](Napi::Env env) mutable {
```

Comment:
Similar issue for the callback capture. No reason to capture if there is no state.

Reply:
fixed